src/ws/session.c

```diff
@@ -378,9 +378,9 @@ acquire_service_credentials (gss_OID mech_type, gss_cred_usage_t usage, gss_cred
 
   if (GSS_ERROR (major))
     {
-      /* This is a routine error message, don't litter */
       const char *msg = gssapi_strerror (mech_type, major, minor);
-      if (!strstr (msg, "nonexistent or empty"))
+      /* don't litter journal with expected error messages if keytab was not set up */
+      if (major != GSS_S_UNAVAILABLE && major != GSS_S_NO_CRED)
```

Comment:
Actually I only observed GSS_S_NO_CRED "in the wild" when /etc/cockpit/krb5.keytab does not exist. So maybe we should restrict it to the latter. Indeed [the docs](https://docs.oracle.com/cd/E19683-01/816-1331/reference-4/index.html) say that `GSS_S_UNAVAILABLE` means "The operation or option is unavailable" which is certainly unexpected.

Reply:
Fixed.

Reply:
Bwah, the original string [still appears in our tests on older OSes](https://logs.cockpit-project.org/logs/pull-16167-20210728-084740-be421b0d-ubuntu-stable/log.html#154). But there it is `GSS_S_FAILURE`, with a totally incomprehensible "minor" code of 0o22661635265. I'll put back the string check for the old releases.